govuk-one-login/di-account-management-backend
# OLH-1052 - Create a new lambda that sends a ticket using the Zendesk API

## Proposed changes

JIRA: https://govukverify.atlassian.net/browse/OLH-1052
Create a new lambda that sends a ticket using the Zendesk API

### What changed
1. New Lambda and SAMs template
2. New Secrets to be added across all envs for interacting with Zendesk

What the Lambda does
1. Ensure all environment variables required to successfully connect to, create ticket and send to Zendesk are provided
2. Validates the fields in the received event
3. Retrieves the values for the environment variable keys in AWS Secrets
4. Validates that values exist for the required keys - these include zendesk api credentials, groups, etc
5. Creates a zendesk ticket and sends to Zendesk
6. If any of the steps above fails, send the SNS record to DLQ for retry later

### Why did it change

So that fraud team are alerted when a suspicious activity is reported and have a way to track/handle all associated investigations and interactions.

### Related links
https://github.com/govuk-one-login/di-account-management-backend/blob/main/docs/adr/0008-report-a-suspicious-activity.md

## Checklists

### Environment variables or secrets
Environment variables
- ZENDESK_GROUP_ID_KEY
- ZENDESK_TAGS_KEY
- ZENDESK_API_TOKEN_KEY
-  ZENDESK_API_USER_KEY
-  ZENDESK_API_URL_KEY

Secrets
- /account-mgmt-backend/zendesk-group-id-token
- /account-mgmt-backend/zendesk-tags-key
- /account-mgmt-backend/zendesk-api-token
- /account-mgmt-backend/zendesk-api-user-key
- /account-mgmt-backend/zendesk-api-url-key

### Permissions


## Testing

1. Deploy to dev environment
2. Drop a suspicious Activity Event to SuspiciousActivityTopic
3. As a user within the 2nd Line Support Group on Zendesk, you should receive an email that a new ticket has been created
4. Navigate to ZenDesk, the 2nd Line support group and view the ticket details

## How to review
Code completeness, test and quality
